relay-server/src/utils/rate_limits.rs

```diff
@@ -677,6 +677,15 @@ where
                 scoping.item(DataCategory::Profile),
                 summary.profile_quantity,
             )?;
+
+            // Profiles can persist in envelopes without transaction if the transaction item
+            // was dropped by dynamic sampling.
+            if profile_limits.is_empty() && summary.event_category.is_none() {
+                profile_limits = self
+                    .check
+                    .apply(scoping.item(DataCategory::Transaction), 0)?;
```

Comment:
We don't need to check for `TransactionIndexed` here, because indexed transactions will trigger the `if let Some(category) = summary.event_category {` branch above.

Reply:
I think you're correct that we don't have to check indexed here, but for the wrong reason.

It won't trigger the branch, because here we never have an `event_category`. But we don't have to check the indexed payload because I assume we don't want to enforce the indexed profile quota on profiles.

But this really is awkward and tricky, because ideally we should persist the information that we dropped indexed transactions on the profile (so it is treated like dynamic sampling), because then the profiling consumers later know that there won't be any transactions for the profiles (like it would be the case if the transaction is dynamically sampled).

As a side note, god damn I hate this function.

Reply:
> It won't trigger the branch, because here we never have an `event_category`.

Let's break it down:

1. If the dynamic sampling decision was "keep" and there is a limit on either `transaction` or `transaction_indexed`, the profile is already dropped [here](https://github.com/getsentry/relay/blob/31b7b8aa94f8bd53dc6f40987f6dffdb485220ea/relay-server/src/utils/rate_limits.rs#L667-L674).
2. If the dynamic sampling decision was "drop"  and there is a limit on `transaction`, the new code drops the profile.
3. If the dynamic sampling decision was "drop" and there is a limit on `transaction_indexed`, we are in undefined territory. I want to err on the side of not enforcing `transaction_indexed` on profiles, but it is awkward like you said, because in the "keep" case we _would_ drop the profile.

> As a side note, god damn I hate this function.

Yes.